### PR TITLE
Switch from retworkx to rustworkx package

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

In the retworkx/rustworkx 0.12.0 release the package was renamed from retworkx to rustworkx which was done at the request of the networkx maintainers [1]. While the retworkx name continues to work for backwards compatibility, it is best to move off of the legacy name and start using the new name sooner rather than later. This commit updates all the retworkx usage to the new rustworkx and updates our requirements list to directly depend on rustworkx.

### Details and comments

Closes #8340 (obsolete)

[1] https://qiskit.org/documentation/rustworkx/release_notes.html#prelude

### CODE REVIEW

File: releasenotes/notes/rustworkx-not-retworkx-b7c4da600df58701.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+upgrade:
+  - |
+    The ``retworkx`` dependency for Qiskit has been removed and replaced by
+    ``rustworkx`` library. These are the same packages, but ``rustworkx`` is
+    the new name for ``retworkx`` which was renamed as part of their combined
+    0.12.0 release. If you were previously using retworkx 0.12.0 with Qiskit
+    then you already were install rustworkx (retworkx 0.12.0 was just a redirect
```
Comment (jakelishman) [inline releasenotes/notes/rustworkx-not-retworkx-b7c4da600df58701.yaml:8]:
```suggestion
    then you already installed rustworkx (retworkx 0.12.0 was just a redirect
```

**Review by jakelishman — COMMENTED** [review]
Ivan (@IvanIsCoding) has old PR #8340 that did this too, but that never merged at the time because it was quite a bit before the rename actually happened.  I don't know what the two of you want to do about the two PRs.

File: qiskit/transpiler/preset_passmanagers/common.py [hunk 2]
```diff
@@ -510,9 +510,9 @@ def get_vf2_call_limit(
...
     if layout_method is None and initial_layout is None:
         if optimization_level == 1:
-            vf2_call_limit = int(5e4)  # Set call limit to ~100ms with retworkx 0.10.2
+            vf2_call_limit = int(5e4)  # Set call limit to ~100ms with rustworkx 0.10.2
         elif optimization_level == 2:
-            vf2_call_limit = int(5e6)  # Set call limit to ~10 sec with retworkx 0.10.2
+            vf2_call_limit = int(5e6)  # Set call limit to ~10 sec with rustworkx 0.10.2
         elif optimization_level == 3:
-            vf2_call_limit = int(3e7)  # Set call limit to ~60 sec with retworkx 0.10.2
+            vf2_call_limit = int(3e7)  # Set call limit to ~60 sec with rustworkx 0.10.2
```
[the remark this answers is not in the document]
Comment (mtreinish) [reply qiskit/transpiler/preset_passmanagers/common.py:517]: Heh, you can tell I just ran sed on the repo. :)

File: qiskit/utils/optionals.py [hunk 3]
```diff
@@ -93,8 +93,8 @@
         be installed in order to use them.
 
     * - .. py:data:: HAS_NETWORKX
-      - Internally, Qiskit uses the high-performance `retworkx
-        <https://github.com/Qiskit/retworkx>`__ library as a core dependency, but sometimes it can
+      - Internally, Qiskit uses the high-performance `rustworkx
+        <https://github.com/Qiskit/rustworkx>`__ library as a core dependency, but sometimes it can
         be convenient to convert things into the Python-only `NetworkX <https://networkx.org/>`__
         format.  There are converter methods on :class:`.DAGCircuit` if NetworkX is present.
```
[the remark this answers is not in the document]
Comment (jakelishman) [reply qiskit/utils/optionals.py:99]: Done in #9165.

**Comment by mtreinish** [comment]
As for this PR vs #8340 I think we probably should just move forward with this one. I had forgotten that @IvanIsCoding had tested this a few months ago, if I had seen that PR I would have just pushed to that branch instead of opening a new PR. But assuming @IvanIsCoding doesn't mind since this version is working and up to date already I think it would better to just merge this and get us migrated to the new name sooner rather than later.

**Review by jakelishman — DISMISSED** [review]
I'm happy to move ahead with merging this.  I'll hold off tagging automerge til Ivan's had a chance to respond.

**Comment by IvanIsCoding** [comment]
> As for this PR vs #8340 I think we probably should just move forward with this one. I had forgotten that @IvanIsCoding had tested this a few months ago, if I had seen that PR I would have just pushed to that branch instead of opening a new PR. But assuming @IvanIsCoding doesn't mind since this version is working and up to date already I think it would better to just merge this and get us migrated to the new name sooner rather than later.

Let's merge Matthew's PR as it is ready, they are the same diff

**Review by jakelishman — APPROVED** [review]

**Pull request merged**